### PR TITLE
Eyes As Organs

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: Eyes are now organs. You can remove or implant them into peoples heads with organ manipulation surgery. A mob without eyes will obviously have trouble seeing.
add: All special eye powers are now tied to their respective organs. For example, this means you can harvest an alien or shadow persons eyes, have them implanted, and gain toggle-able night vision.
add: All cybernetic eye implants are now cybernetic eyes, meaning you must replace the patients organic eyes. HUD implants are still just regular implants.
/:cl:

**Done:**

Eye organ that holds all the information for dark sight, see invisible, vision flags etc that previously was tied to species or mob

Making humans spawn with those eyes

Losing vision on losing eyes/getting a missing eyes overlay

Killing the terrible copypaste across all carbon update_sights()

Aliens, shadow people both have their own eyes that can toggle night vision.

Eye implants converted to eye organs

Eye surgery updated to take eyes into account

Eyes travel with dismembered heads properly

**To do in later PRs:**

Tie eye damage (welding, pepperspray) to eye organs

Tie DNA eye color to eye organs?

### CODE REVIEW

**Comment by dannno** [comment]
give flashlight eyes the thing i made or else

**Comment by Cheridan** [comment]
TODO: replace only one eye so i can be a anime heterochromia snowflake

**Comment by XDTM** [comment]
Aren't eye implants already cybernetic organs?

**Comment by KorPhaeron** [comment]
Yeah but they don't replace your eyes right now, you just shove them in peoples heads

**Comment by KorPhaeron** [comment]
Like arm implants go in the arm spot and robotic arms go in the arm slot but only one actually replaces the arm

**Comment by XDTM** [comment]
I think it makes sense to have them work as HUD lenses/shielding that go over your real eye, especially since they stack right now. New eye implants could definitely replace real eyes, though.

**Comment by PKPenguin321** [comment]
Xray and thermal eyes should replace the whole eye, but HuDs should stay as weird little overlay things

**Comment by XDTM** [comment]
Agree on those. Welding shield should be separate.

**Comment by KorPhaeron** [comment]
I think vague "implants" that don't really interact with the body at all other than being in them or not are kind of boring vs getting to actually remove and replace parts though

**Comment by XDTM** [comment]
I definitely agree, but i'd rather not make them unstackable and make new implants instead.

**Comment by TehZombehz** [comment]
Does this update the regenerate_organs proc to regenerate the eye organs, too?

**Comment by KorPhaeron** [comment]
@TehZombehz 

It does now, thanks

**Comment by Kromgar** [comment]
So... is the eye of god going to require surgery now? Or will it just replace one of your eyes?

**Comment by KorPhaeron** [comment]
@tgstation/commit-access 

Ready for review please

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 1]
```diff
@@ -1,8 +1,12 @@
 
 /mob/living/carbon/get_eye_protection()
 	var/number = ..()
-	for(var/obj/item/organ/cyberimp/eyes/EFP in internal_organs)
-		number += EFP.flash_protect
+
+	var/obj/item/organ/eyes/E = getorganslot("eye_sight")
+	if(!E)
+		number = INFINITY //Can't get flashed without eyes
```
Comment (optimumtact) [inline code/modules/mob/living/carbon/carbon_defense.dm]: new meta

File: code/modules/mob/living/simple_animal/hostile/statue.dm [hunk 2]
```diff
@@ -206,7 +206,8 @@
 	include_user = 1
 
 /obj/effect/proc_holder/spell/targeted/night_vision/cast(list/targets,mob/user = usr)
-	for(var/mob/living/target in targets)
+	return
+	/*for(var/mob/living/target in targets)
```
Comment (optimumtact) [inline code/modules/mob/living/simple_animal/hostile/statue.dm]: why is there commented out code here

**Comment by optimumtact** [comment]
Couple minor comments but it looks fine

**Comment by FantasticFwoosh** [comment]
> Give skeletons googly eyes or add the loot eyes of god as a blood-demon heart esque eye organ when?

**Comment by XDTM** [comment]
Eyes of god are NODROP glasses right now, but i guess they could easily become self-implantable organs

File: code/modules/surgery/eye_surgery.dm [hunk 3]
```diff
@@ -11,6 +11,12 @@
 	implements = list(/obj/item/weapon/hemostat = 100, /obj/item/weapon/screwdriver = 45, /obj/item/weapon/pen = 25)
 	time = 64
 
+/datum/surgery/eye_surgery/can_start(mob/user, mob/living/carbon/target)
+	var/obj/item/organ/eyes/E = target.getorganslot("eye_sight")
+	if(!E)
+		user << "It's hard to do surgery on someones eyes when they don't have any."
+		return 0
```
Comment (lzimann) [inline code/modules/surgery/eye_surgery.dm]: return FALSE

Comment (XDTM) [inline code/modules/surgery/eye_surgery.dm]: Shouldn't there be a check that makes it not appear in the list at all?

File: code/modules/mob/living/simple_animal/hostile/statue.dm (hunk 2 above)
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/hostile/statue.dm]: Forgot to go back and finish this

**Comment by KorPhaeron** [comment]
@optimumtact 

Should be good now

**Comment by XDTM** [comment]
Remove the user << message from the can_start surgery proc, it's not needed since it automatically removes it from the list

File: code/game/gamemodes/changeling/powers/augmented_eyesight.dm [hunk 4]
```diff
@@ -12,54 +12,25 @@
...
 		return
-	if(user.getorgan(/obj/item/organ/cyberimp/eyes/thermals/ling))
-		user << "<span class='notice'>Our eyes are protected from flashes.</span>"
-		var/obj/item/organ/cyberimp/eyes/O = new /obj/item/organ/cyberimp/eyes/shield/ling()
-		O.Insert(user)
-
+	var/obj/item/organ/eyes/E = user.getorganslot("eye_sight")
+	if(E)
+		if(E.flash_protect)
+			E.sight_flags |= SEE_MOBS
```
Comment (RemieRichards) [inline code/game/gamemodes/changeling/powers/augmented_eyesight.dm]: so:
1. activate augmented eyesight power
2. remove own eyes (potentially use a friend)
3. turn the power off
4. insert the new eyes
5. permanent version of the power, respec to get around it.

are my fears unfounded?

Comment (KorPhaeron) [inline code/game/gamemodes/changeling/powers/augmented_eyesight.dm]: They're not unfounded, but using a changeling as an eyeball factory seemed silly and fun enough to leave in

Comment (RemieRichards) [inline code/game/gamemodes/changeling/powers/augmented_eyesight.dm]: Even if they killed eachother during surgery, they're lings they could just res, are eyes removable from head items? if not they should be but that would also be a concern as like how robos just decap people for their brains, lings would just repeatedly decap eachother for mad eyeball profit.

**Comment by FantasticFwoosh** [comment]
You could probably use the ability to print off eyes from a organ grower too rather than brave the space chinese black market.

Emag the machine to print off short sighted eyes/abnormal meson but flash weak ling eyes you can't turn off.

File: code/game/gamemodes/changeling/powers/augmented_eyesight.dm (hunk 4 above)
Comment (KorPhaeron) [inline code/game/gamemodes/changeling/powers/augmented_eyesight.dm]: Don't know if there is really a sane way to fix this sing changelings, like many other things in the game, violate the conservation of energy and just generate stuff out of thin air

**Pull request merged**